wrapanapi/lenovo.py

```diff
@@ -1,42 +1,309 @@
 # coding: utf-8
 """Backend management system classes
-
 Used to communicate with providers without using CFME facilities
 """
+import json
+from urlparse import urlunparse
+
+import requests
+from requests.exceptions import Timeout
+
 from base import WrapanapiAPIBase
 
 
 class LenovoSystem(WrapanapiAPIBase):
     """Client to Lenovo API
-
     Args:
         hostname: The hostname of the system.
         username: The username to connect with.
         password: The password to connect with.
-
     """
     _api = None
 
     _stats_available = {
-        'num_server': lambda self: len(self.list_server()),
+        'num_server': lambda self: len(self.list_servers()),
+        # 'num_server_with_host': lambda self: len(self.list_servers_with_host()),
     }
+    POWERED_ON = 8
+    POWERED_OFF = 5
+    STANDBY = 18
+    HEALTH_VALID = ("normal", "non-critical")
+    HEALTH_WARNING = ("warning")
+    HEALTH_CRITICAL = ("critical", "minor-failure", "major-failure", "non-recoverable", "fatal")
 
-    def __init__(self, hostname, username, password, **kwargs):
-        super(LenovoSystem, self).__init__(kwargs)
+    def __init__(self, hostname, username, password, protocol="https", port=443, **kwargs):
         self.hostname = hostname
         self.username = username
         self.password = password
+        self.protocol = protocol
+        self.port = port
 
     def __del__(self):
         """Disconnect from the API when the object is deleted"""
         # This isn't the best place for this, but this class doesn't know when it is no longer in
         # use, and we need to do some sort of disconnect based on the pyVmomi documentation.
-        raise NotImplementedError
+
+    def _service_instance(self, path):
```

Comment:
can we take a look at simply handing it the full url + auth instead, and perhaps opening a own requests session for the object

in particular the repeated unparsing and auth object construction doesnt sit well with me, a correctly configured session has a auth object that belongs to a url

Reply:
Thanks @RonnyPfannschmidt , I agree with you and I already thinking make some changes.

Reply:
Done @RonnyPfannschmidt